src/include/migraphx/op/pooling.hpp

```diff
@@ -38,19 +38,27 @@ struct pooling
 
     std::string name() const { return "pooling"; }
 
-    shape compute_shape(std::vector<shape> inputs) const
+    void check_attribute_size() const
     {
-        check_shapes{inputs, *this}.has(1);
         if(not(padding.size() == stride.size() and padding.size() == lengths.size()))
         {
             MIGRAPHX_THROW("pooling: inconsistent attribute sizes");
         }
+    }
+
+    shape compute_shape(std::vector<shape> inputs) const
+    {
+        check_shapes{inputs, *this}.has(1);
 
```

Comment:
Looks like we missed the function call check_attribute_size() here

Reply:
it is implicitly called when doing this->kdims() in L58